src_c/line.c

```diff
@@ -490,6 +490,85 @@ pg_line_flip_ip(pgLineObject *self, PyObject *_null)
     Py_RETURN_NONE;
 }
 
+double 
+_lerp_helper(float start, float end, float amount) {
+    return (1 - amount) * start + amount * end;
+}
+
+static int
+_line_scale_helper(pgLineBase *line, double factor, double origin) {
+    if (factor == 1.0) {
+        return 1;
+    }
+    else if (factor <= 0.0) {
+        PyErr_SetString(PyExc_ValueError, "Can only scale by a positive non zero number");
+        return 0;
+    }
+
+    if (origin < 0.0 || origin > 1.0) {
+        PyErr_SetString(PyExc_ValueError, "Origin must be between 0 and 1");
+        return 0;
+    }
+
+    double x1 = line->x1;
+    double y1 = line->y1;
+    double x2 = line->x2;
+    double y2 = line->y2;
+
+    double x1_factor = x1 * factor;
+    double y1_factor = y1 * factor;
+    double x2_factor = x2 * factor;
+    double y2_factor = y2 * factor;
+
+    double fac_m_one = factor - 1;
+    double dx = _lerp_helper(fac_m_one * x1, fac_m_one * x2, origin);
+    double dy = _lerp_helper(fac_m_one * y1, fac_m_one * y2, origin);
+    
+    line->x1 = x1_factor - dx;
+    line->y1 = y1_factor - dy;
+    line->x2 = x2_factor - dx;
+    line->y2 = y2_factor - dy;
```

Comment:
Try to see if this works, might be a simplification if so:
```suggestion
    double dx = _lerp_helper((x1 - x2) * factor, x2 - x1, origin);
    double dy = _lerp_helper((y1 - y2) * factor, y2 - y1, origin);

    line->x1 = x1 + dx;
    line->y1 = y1 + dy;
    line->x2 = x2 + dx;
    line->y2 = y2 + dy;
```